x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/form/serializer/serializer.ts

```diff
@@ -0,0 +1,161 @@
+        }
+
+        if (hotPhaseActions.rollover.max_size) {
+          hotPhaseActions.rollover.max_size = `${hotPhaseActions.rollover.max_size}${_meta.hot.maxStorageSizeUnit}`;
+        }
+
+        if (!updatedPolicy.phases.hot!.actions?.forcemerge) {
+          delete hotPhaseActions.forcemerge;
+        } else if (_meta.hot.bestCompression) {
+          hotPhaseActions.forcemerge!.index_codec = 'best_compression';
+        }
+
+        if (_meta.hot.bestCompression && hotPhaseActions.forcemerge) {
+          hotPhaseActions.forcemerge.index_codec = 'best_compression';
+        }
+      } else {
+        delete hotPhaseActions.rollover;
+        delete hotPhaseActions.forcemerge;
+      }
+
+      if (!updatedPolicy.phases.hot!.actions?.set_priority) {
+        delete hotPhaseActions.set_priority;
+      }
+    }
+
+    /**
+     * WARM PHASE SERIALIZATION
+     */
+    if (_meta.warm.enabled) {
+      const warmPhase = draft.phases.warm!;
+      // If warm phase on rollover is enabled, delete min age field
+      // An index lifecycle switches to warm phase when rollover occurs, so you cannot specify a warm phase time
+      // They are mutually exclusive
+      if (
+        (!_meta.hot.useRollover || !_meta.warm.warmPhaseOnRollover) &&
+        updatedPolicy.phases.warm!.min_age
+      ) {
+        warmPhase.min_age = `${updatedPolicy.phases.warm!.min_age}${_meta.warm.minAgeUnit}`;
+      } else {
+        delete warmPhase.min_age;
+      }
+
+      warmPhase.actions = serializeMigrateAndAllocateActions(
+        _meta.warm,
+        warmPhase.actions,
+        originalPolicy?.phases.warm?.actions
+      );
+
+      if (!updatedPolicy.phases.warm!.actions?.forcemerge) {
+        delete warmPhase.actions.forcemerge;
+      } else if (_meta.warm.bestCompression) {
+        warmPhase.actions.forcemerge!.index_codec = 'best_compression';
+      }
+
+      if (!updatedPolicy.phases.warm!.actions?.set_priority) {
+        delete warmPhase.actions.set_priority;
+      }
+
+      if (!updatedPolicy.phases.warm!.actions?.shrink) {
+        delete warmPhase.actions.shrink;
+      }
+    } else {
+      delete draft.phases.warm;
+    }
+
+    /**
+     * COLD PHASE SERIALIZATION
+     */
+    if (_meta.cold.enabled) {
+      const coldPhase = draft.phases.cold!;
+
+      if (updatedPolicy.phases.cold!.min_age) {
+        coldPhase.min_age = `${updatedPolicy.phases.cold!.min_age}${_meta.cold.minAgeUnit}`;
+      }
+
+      coldPhase.actions = serializeMigrateAndAllocateActions(
+        _meta.cold,
+        coldPhase.actions,
+        originalPolicy?.phases.cold?.actions
+      );
+
+      if (_meta.cold.freezeEnabled) {
+        coldPhase.actions.freeze = coldPhase.actions.freeze ?? {};
+      } else {
+        delete coldPhase.actions.freeze;
+      }
+
+      if (!updatedPolicy.phases.cold!.actions?.set_priority && coldPhase.actions.set_priority) {
+        delete coldPhase.actions.set_priority;
```

Comment:
This line too.

This is another line I don't see corresponding logic for in the original. Is it new?